### PR TITLE
Log Exception in LoggingX509ExtendedTrustManager

### PR DESCRIPTION
**Pull request by Hakky54**

### CODE REVIEW

File: sslcontext-kickstart/src/main/java/nl/altindag/ssl/trustmanager/LoggingX509ExtendedTrustManager.java [hunk 1]
```diff
@@ -38,57 +38,69 @@ public final class LoggingX509ExtendedTrustManager extends DelegatingX509Extende
...
         String logMessage = String.format(LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""), counterParty, certificateChain);
         LOGGER.debug(logMessage);
+
+        try {
+            runnable.run();
+            String okMessage = String.format(VALIDATION_PASSED_LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""));
+            LOGGER.debug(okMessage);
+        } catch (CertificateException e) {
+            String nokMessage = String.format(VALIDATION_FAILED_LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""));
+            LOGGER.debug(nokMessage, e);
```
Comment (maxxedev) [inline sslcontext-kickstart/src/main/java/nl/altindag/ssl/trustmanager/LoggingX509ExtendedTrustManager.java:101]: can you make it possible for users to enable logging of certificate exception but not successful check?

some possibilities:
1. different loggers:
```
Logger SUCCESS_LOGGER = getLogger(LoggingX509ExtendedTrustManager.class.getName() + ".success");
Logger SUCCESS_LOGGER = getLogger(LoggingX509ExtendedTrustManager.class.getName() + ".exception");
```

2. slf4j markers

3. allow slf4j level to be passed:
```
TrustManagerUtils.createLoggingTrustManager(delegate,
    org.slf4j.event.Level.DEBUG, // level for success
    org.slf4j.event.Level.WARN); // level for exception
```

4. protected methods for subclass overriding

Comment (Hakky54) [reply sslcontext-kickstart/src/main/java/nl/altindag/ssl/trustmanager/LoggingX509ExtendedTrustManager.java:101]: I think the first option would be the better one. In that way you can disable the part which you don't want in your logging configuration file

File: sslcontext-kickstart/src/main/java/nl/altindag/ssl/trustmanager/LoggingX509ExtendedTrustManager.java [hunk 2]
```diff
@@ -37,58 +37,73 @@
...
         String logMessage = String.format(LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""), counterParty, certificateChain);
         LOGGER.debug(logMessage);
+
+        try {
+            runnable.run();
+            String okMessage = String.format(VALIDATION_PASSED_LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""));
+            SUCCESS_LOGGER.debug(okMessage);
+        } catch (CertificateException e) {
+            String nokMessage = String.format(VALIDATION_FAILED_LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""));
+            EXCEPTION_LOGGER.debug(nokMessage, e);
```
Comment (maxxedev) [inline sslcontext-kickstart/src/main/java/nl/altindag/ssl/trustmanager/LoggingX509ExtendedTrustManager.java:104]: how can users customize the message? should that be possible?

Comment (Hakky54) [reply sslcontext-kickstart/src/main/java/nl/altindag/ssl/trustmanager/LoggingX509ExtendedTrustManager.java:104]: You mean the content or the visibility?

Comment (maxxedev) [reply sslcontext-kickstart/src/main/java/nl/altindag/ssl/trustmanager/LoggingX509ExtendedTrustManager.java:104]: both

Comment (Hakky54) [reply sslcontext-kickstart/src/main/java/nl/altindag/ssl/trustmanager/LoggingX509ExtendedTrustManager.java:104]: The message cannot be adjusted. The visibility can be adjusted. The library  has slf4j api, so you need to add a logger implementation such as logback or slf4j simple and configure your logs with it

**Pull request merged**